docs/roadmap.md

```diff
@@ -0,0 +1,112 @@
+# Roadmap
+
+* [Core Functionality Bug Fixes](#core-functionality-bug-fixes)
+* [2000+ Applications support](#2000-applications-support)
+* [100+ Clusters support](#100-clusters-support)
+* [Mono Repository support](#mono-repository-support)
+* [ApplicationsSet](#applicationsset)
+* [Large Applications support](#large-applications-support)
+* [Supportability](#supportability)
+* [GitOps Engine Enhancements](#gitops-engine-enhancements)
+* [GitOps Agent](#gitops-agent)
+* [Config Management Tools integrations](#config-management-tools-integrations)
+* [Cluster Management User interface](#cluster-management-user-interface)
+* [Resource Actions revamp](#resource-actions-revamp)
+* [Argo CD Notifications](#argo-cd-notifications)
+* [Automated Registry Monitoring](#automated-registry-monitoring)
+* [Application details page usability](#application-details-page-usability)
+* [Projects enhancements](#projects-enhancements)
+
+### Core Functionality Bug Fixes
+
+The core GitOps features still have several known bugs and limitations. The full list is available in [v1.7 milestone](
+https://github.com/argoproj/argo-cd/issues?q=is%3Aopen+is%3Aissue+label%3Abug+milestone%3A%22v1.7+%22+label%3Acomponent%3Acore)
+
+The most notable issues:
+* [Application is incorrectly reporting a diff](https://github.com/argoproj/argo-cd/issues/2865)
+* [Helm hooks are deleted right after creation](https://github.com/argoproj/argo-cd/issues/2737)
+* [Argo CD synchronization lasts incredibly long](https://github.com/argoproj/argo-cd/issues/3663)
+
+### Performance
+
+* 2000+ Applications support. The user interface became notable slower if one Argo CD instance manages more than 1 thousand applications.
+A set of optimizations is required to fix that issue.
+
+* 100+ Clusters support. The cluster addon management use-case requires connecting a large number of clusters to one Argo CD controller.
+Currently Argo CD controller unable to handle that many clusters. The solution to support horizontal controller scaling and automated sharding.
```

Comment:
*Currently Argo CD controller _is_ unable to...

Reply:
*The solution _is_ to

Reply:
thank you!

Reply:
done.